src/exercises/common.mk

```
@@ -0,0 +1,26 @@
+CC=clang
+
+ARCH!=uname -m
+.if ${ARCH} == "arm64"
+CFLAGS=-g -O2 -target aarch64-unknown-freebsd -Wall -Wcheri
+CFLAGS_CHERI=-march=morello+noa64c -march=morello -mabi=purecap -Xclang -morello-vararg=new
+.else
+CFLAGS=-g -O2 -target riscv64-unknown-freebsd -Wall -Wcheri -mno-relax
+CFLAGS_CHERI=-march=rv64gcxcheri -mabi=l64pc128d 
+.endif
```

```suggestion
CC=cc
CFLAGS=-g -O2 -Wall -Wcheri

ARCH!=uname -m
.if ${ARCH} == "arm64"
CFLAGS_BASELINE=-march=morello+noa64c -mabi=aapcs
CFLAGS_CHERI=-march=morello -mabi=purecap
.else
CFLAGS_BASELINE=-march=rv64gc -mabi=lp64d
CFLAGS_CHERI=-march=rv64gcxcheri -mabi=l64pc128d
.endif
```

(CFLAGS_CHERI could be empty on purecap systems since cc defaults to the native ABI, but this is probably better practice)